src/on.ts

```diff
@@ -0,0 +1,97 @@
+import { Handle, EventObject } from './interfaces';
+import { createHandle, createCompositeHandle } from './lang';
+import Evented from './Evented';
+
+// Only used for Evented and EventEmitter, as EventTarget uses EventListener
+type EventCallback = (event: {}) => void;
+
+export interface ExtensionEvent {
+	(target: any, listener: EventCallback): Handle;
+}
+
+export interface EventTarget {
+	accessKey?: string;
+	addEventListener(event: string, listener: EventListener, capture?: boolean): void;
+	removeEventListener(event: string, listener: EventListener, capture?: boolean): void;
+}
+
+export interface EventEmitter {
+	on(event: string, listener: EventCallback): EventEmitter;
+	removeListener(event: string, listener: EventCallback): EventEmitter;
+}
+
+export function emit(target: EventTarget, event: EventObject): boolean;
+export function emit(target: EventEmitter, event: EventObject): boolean;
+export function emit(target: Evented, event: EventObject): boolean;
+
+export function emit(target: any, event: any): boolean {
+	if (target.dispatchEvent && target.ownerDocument && target.ownerDocument.createEvent) {
+		let nativeEvent = target.ownerDocument.createEvent('HTMLEvents');
+		nativeEvent.initEvent(event.type, Boolean(event.bubbles), Boolean(event.cancelable));
+
+		for (let key in event) {
+			if (!(key in nativeEvent)) {
+				nativeEvent[key] = event[key];
+			}
+		}
+
+		return target.dispatchEvent(nativeEvent);
+	}
+
+	if (target.emit && target.removeListener) {
+		return target.emit(event.type, event);
```

Comment:
This depends on our decision to normalize return values of `emit`. Do we pave over `EventEmitter.emit`'s functionality that returns `true` or `false` depending on if callbacks were registered? For now I left this in, but calling `target.emit(event.type, event);` then explicitly returning `false` would be fine if that's preferable.